packages/jest-dom-mocks/src/document.ts

```diff
@@ -0,0 +1,42 @@
+export interface MockWindow extends Window {
+  addEventListener: jest.Mock<{}> & typeof window.addEventListener;
+  removeEventListener: jest.Mock<{}> & typeof window.removeEventListener;
+}
+
+export default class Document {
+  private isUsingMockWindow: boolean = false;
+  private window?: MockWindow;
+
+  mockWindow() {
+    window.addEventListener = jest.fn();
+    window.removeEventListener = jest.fn();
+    return window as MockWindow;
+  }
+
+  mock() {
+    if (this.isUsingMockWindow) {
+      throw new Error('You tried to mock window when it was already mocked.');
+    }
+
+    this.window = this.mockWindow();
+    this.window.addEventListener('load', () => {});
```

Comment:
Overriding `window.addEventListener` does not mean you are also overriding `document.addEventListener` (since you are just overriding a reference, not the function itself). Considering we use `document.addEventListener` *and* `window.addEventListener` in our codebases, we probably want to override both of these.

**example**
![image](https://user-images.githubusercontent.com/4889856/45452194-55cd9780-b6ab-11e8-811f-0debe311cde1.png)